src/views/ManageList.jsx

```diff
@@ -31,10 +30,39 @@ export function ManageList({ userId }) {
 		const listPath = localStorage.getItem('tcl-shopping-list-path');
 
 		if (!listPath) {
-			window.alert('List is not existed.');
+			window.alert('List does not exist.');
 			return;
 		}
 
+		const formDataCheck = formData.name
+			.replace(/[^A-Z0-9]/gi, '')
+			.toLowerCase();
+
+		if (
+			data.find(
+				(item) =>
+					item.name.trim().toLowerCase() === formData.name.trim().toLowerCase(),
+			)
+		) {
+			window.alert(`${formData.name} already exists in your list.`);
+			return;
+		}
+
+		for (const item of data) {
+			const itemCheck = item.name.replace(/[^A-Z0-9]/gi, '').toLowerCase();
+			if (itemCheck === formDataCheck || formDataCheck.includes(itemCheck)) {
+				if (
+					!window.confirm(
+						`A similar item is already on your list. Do you still want to add ${formData.name}?`,
+					)
+				) {
+					return;
+				} else {
+					break;
+				}
+			}
+		}
```

Comment:
<!-- Suggestions propose improvements to the current subject.  -->
**suggestion ( non-blocking ):** This 100% works, so no need to change it, but it might be interesting to look at the [`.some` method in MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). It won't break early if it finds a match, but with the amount of data we're looking at, the time it saves is negligible anyhow and it would make the code easier to read.

Reply:
Good point, just added it and it works just fine! Broke the some into two lines to not have to repeat regex on the `item.name `(I cant help it 😅). But still a lot more readable, thank you!